elastic/kibana
# [Infra]  Improve hosts onboarding flow

closes [#189172](https://github.com/elastic/kibana/issues/189172)
closes https://github.com/elastic/observability-dev/issues/3226 (internal)

## Summary

Introduces a better onboarding flow for Hosts to the Hosts View and Asset Details page. Other entity types and infra pages remain showing the current flow.

**Hosts View**

https://github.com/user-attachments/assets/2fbd9534-05ef-4ec0-848b-76a1733682b8

https://github.com/user-attachments/assets/4cc2e04f-644e-4b3c-b6fb-810a664bae1a

**Asset Details**

https://github.com/user-attachments/assets/c093475f-625f-4d51-a6f7-8696757bd9e4


### Extra

Removed the "Loading data sources" spinner

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/8b769abc-fc26-408d-a531-806035577807">


| before| after |
|-------|------|
| ![with_loading_sources](https://github.com/user-attachments/assets/bb76f128-6304-426a-bade-b847a9bddc25) | ![without_loading_sources](https://github.com/user-attachments/assets/974069da-eea0-475a-b698-5d7e79996bd1)  |


### How to test

- Start a local Kibana and es instances
- Run any APM synthtrace scenario `node scripts/synthtrace simple_trace.ts --live`
- Navigate to Infrastructure > Hosts
    - The no data prompt should be displayed, the "add data" button should redirect to `observabilityOnboarding/?category=logs`

- Run synthtrace scenario with system integration data `node scripts/synthtrace infra_hosts_with_apm_hosts --live`
- Navigate to Infrastructure > Hosts
- Click on "Add data" at the top menu
  - it should redirect to `observabilityOnboarding/?category=logs`
- Return to the hosts view and click on a host to navigate to the asset details page 
- Click on "Add data" at the top menu
  - it should redirect to `observabilityOnboarding/?category=logs`

Inventory UI, Metrics UI, Asset details (containers) and legacy node details page should all point to the current onboarding flow.
